### PR TITLE
fix(rds): Make most DatabaseClusterAttributes properties optional

### LINKED ISSUE #3587: All properties of DatabaseClusterAttributes should not be required
**Note: for support questions, please first reference our [documentation](https://docs.aws.amazon.com/cdk/api/latest), then use [Stackoverflow](https://stackoverflow.com/questions/ask?tags=aws-cdk)**. This repository's issues are intended for feature requests and bug reports.

* https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseClusterAttributes.html
* https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-rds.DatabaseCluster.html

* **I'm submitting a ...**
  - [x] :beetle: bug report
  - [ ] :rocket: feature request
  - [ ] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/aws/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*

`DatabaseCluster.from­Database­Cluster­Attributes()` takes the `DatabaseClusterAttributes` interface as a props argument for filtering purposes. But, since every property of the interface is required, the method `from­Database­Cluster­Attributes()` not very useful.

* **What is the expected behavior (or behavior of feature suggested)?**

All properties of DatabaseClusterAttributes should be optional except for a few essential ones, if any.

* **What is the motivation / use case for changing the behavior or adding this feature?**

Currently, there is no way to refer to existing DB from CDK app. `DatabaseCluster.from­Database­Cluster­Attributes()` is the only one I'm aware of, but I cannot use it because all fields are required.

* **Please tell us about your environment:**

  - CDK CLI Version: 1.3.0 (build bba9914)
  - Module Version: 1.3.0
  - OS: OSX Mojave
  - Language: TypeScript

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

### PR DESCRIPTION
**Pull request by njlynch**
The `IDatabaseCluster` interface has a lot of properties, putting a large strain
on users who want to import clusters via `DatabaseClusterAttributes`. This
alters the attributes interface to contain almost exclusively optional props,
and lazily throw if any missing properties are accessed.

I believe this is a backwards-compatible change, but someone please double-check
me on that.

fixes #3587

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by skinny85 — APPROVED** [review]
Love it!

Leaving it with `do-not-merge` to let @shivlaks chime in, but I'm totally fine with pushing this as-is.

**Review by shivlaks — COMMENTED** [review]
LGTM as well!

File: packages/@aws-cdk/aws-rds/lib/cluster-ref.ts [hunk 1]
```diff
@@ -46,38 +46,50 @@ export interface IDatabaseCluster extends IResource, ec2.IConnectable, secretsma
  */
 export interface DatabaseClusterAttributes {
   /**
-   * The database port
+   * Identifier for the cluster
    */
-  readonly port: number;
+  readonly clusterIdentifier: string;
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-rds/lib/cluster-ref.ts:51]: any reason we wouldn't want to just call it `clusterId` - don't think we expand it out to `identifier` in many places do we?

File: packages/@aws-cdk/aws-rds/lib/cluster-ref.ts [hunk 2]
```diff
@@ -46,38 +46,50 @@ export interface IDatabaseCluster extends IResource, ec2.IConnectable, secretsma
...
-  readonly clusterIdentifier: string;
+  readonly securityGroups?: ec2.ISecurityGroup[];
 
   /**
    * Identifier for the instances
+   *
+   * @default - no instance identifiers
    */
-  readonly instanceIdentifiers: string[];
+  readonly instanceIdentifiers?: string[];
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-rds/lib/cluster-ref.ts:72]: was this why we went with `clusterIdentifier` for the property name, or was there another reason?

**Pull request merged**